### PR TITLE
Replace set and state keywords with new statement endings, sequencing op

### PR DESCRIPTION
**Pull request by akalicki**
Resolves issue #36.
 - remove set and state
 - use `;` as statement end
 - use `and` as sequencer
 - use `return expr;` to end functions

### CODE REVIEW

File: test/scanner/_evenfunc.in [hunk 1]
```diff
@@ -1,2 +1,2 @@
-set iseven = (num) -> 
-  return if num % 2 == 0 then true else "false"
\ No newline at end of file
+iseven = (num) -> 
+  return if num % 2 == 0 then true else "false";
```
Comment (akalicki) [inline test/scanner/_evenfunc.in]: can someone verify this is how we'd write functions?

File: test/parser/parserize.ml [hunk 2]
```diff
@@ -19,10 +19,10 @@ let rec txt_of_expr = function
     let v1 = txt_of_expr e1 and op1 = txt_of_op op and v2 = txt_of_expr e2 in
     "Binop(" ^ v1 ^ ", " ^ op1 ^ ", " ^ v2 ^ ")"
   | Call(f, args) -> let args1 = List.map txt_of_expr args in
-    "Call(Id(" ^ f ^ "), [" ^ String.concat " ; " args1 ^ "])"
```
Comment (lillyfwang) [inline test/parser/parserize.ml]: Why was "Id" removed? Was it a mistake to have it in there in the first place?

Comment (akalicki) [inline test/parser/parserize.ml]: right now we're parsing functions as `Call of string * expr list`, not with IDs. We might eventually want to parse the function name as an ID and not a string though

File: compiler/generator.ml [hunk 3]
```diff
@@ -47,7 +47,7 @@ and txt_of_args = function
   | _ as arg_list -> String.concat ", " (List.map txt_of_expr arg_list)
 
 let process_stmt = function
-  | State(expr) -> sprintf "%s" (txt_of_expr expr)
+  | Expr(expr) -> sprintf "%s" (txt_of_expr expr)
```
Comment (lillyfwang) [inline compiler/generator.ml]: Do()? Expr()?

File: compiler/parser.mly [hunk 4]
```diff
@@ -11,7 +11,10 @@
 %{ open Ast %}
 
 /* Punctuation */
-%token LPAREN RPAREN LCAR RCAR LBRACK RBRACK SEMI COMMA VBAR
+%token LPAREN RPAREN LCAR RCAR LBRACK RBRACK COMMA VBAR
+
+/* Sequencing */
+%token SEQ
```
Comment (lillyfwang) [inline compiler/parser.mly]: Will we need this? Or is this a convo for a new branch/commit

**Pull request merged**